tests/test_parser_ca.py

```diff
@@ -270,152 +239,33 @@ def test_building(input_data, expected):
         ("Suite 1800 ", True),
         ("Apt 1B ", True),
         ("Rm. 52 ", True),
-        ("#2b ", True),
     ],
 )
-def test_occupancy_positive(input_data, expected):
+def test_occupancy(input_data: str, match_expectation: bool) -> None:
     """tests exact string match for a place id"""
-    match = utils.match(data_ca.occupancy, input_data, re.VERBOSE)
-    is_found = match is not None
-    assert is_found == expected
-    assert match.group(0) == input_data
+    execute_matching_test(input_data, match_expectation, data_ca.occupancy)
 
 
 @pytest.mark.parametrize(
-    ("input_data", "expected"),
-    [
-        # positive assertions
-        ("suite900 ", False),
-        ("Suite#2", False),
-        ("suite218 ", False),
-    ],
-)
-def test_occupancy_negative(input_data, expected):
-    """tests string match for a place id"""
-    match = utils.match(data_ca.occupancy, input_data, re.VERBOSE)
-    is_found = match is not None
-    assert is_found == expected
-
-
-@pytest.mark.parametrize(
-    ("input_data", "expected"),
+    ("input_data", "match_expectation"),
     [
         # positive assertions
         ("po box 108", True),
         ("Po Box 53485", True),
         ("P.O. box 119", True),
         ("PO box 1070", True),
+        # negitive assertions
+        ("POb ox1070", False),
+        ("boxer 123", False),
     ],
 )
-def test_po_box_positive(input_data, expected):
-    """tests exact string match for a po box"""
-    match = utils.match(data_ca.po_box, input_data, re.VERBOSE)
-    is_found = match is not None
-    assert is_found == expected
-    assert match.group(0) == input_data
-
-
-@pytest.mark.parametrize(
-    ("input_data", "expected"),
-    [
-        # positive assertions
-        ("po box108 ", False),
-        ("PoBox53485 ", False),
-        ("P.O. box119", False),
-        ("POb ox1070 ", False),
-    ],
-)
-def test_po_box_negative(input_data, expected):
+def test_po_box(input_data: str, match_expectation: bool) -> None:
     """tests string match for a po box"""
-    match = utils.match(data_ca.po_box, input_data, re.VERBOSE)
-    is_found = match is not None
-    assert is_found == expected
-
-
-"""
-NOTE:
-Testing for 'full_street' below is meaningless
-since "full_street_b" regexp is based on positive
-lookahead assertion and lazy regex before it,
-so it will fail most of the tests below, while
-still finding correct matches in full_address
-"""
+    execute_matching_test(input_data, match_expectation, data_ca.po_box)
 
 
 @pytest.mark.parametrize(
-    ("input_data", "expected"),
-    [
-        # positive assertions
-        ("15979 Bow Bottom Trail SE, Calgary, AB T2J 6T5", True),
```

Comment:
why get rid of these?

Reply:
cos of the above comment? why keep meaningless tests
```
"""
NOTE:
Testing for 'full_street' below is meaningless
since "full_street_b" regexp is based on positive
lookahead assertion and lazy regex before it,
so it will fail most of the tests below, while
still finding correct matches in full_address
"""
```

Reply:
wouldn't it be good to have expected failures instead? 

I'm not really that bothered either way